src/clj_irods/core.clj

```diff
@@ -343,3 +380,67 @@
            (jargon/cached-get-path irods uuid)
            (when (:has-jargon irods)
              (jargon/get-path irods uuid)))) uuid])))
+
+(defn uuids->paths
+  "The paths associated with multiple uuids, returned as a map from uuid to
+  the corresponding path or nil"
+  [irods uuids]
+  (otel/with-span [s ["uuids->paths"]]
+    (cached-or-get irods
+      [(fn [cache? irods uuids]
+         (when (:has-icat irods)
+           (icat/paths-for-uuids irods uuids))) uuids]
+      [(fn [cache? irods uuids]
+         (when (:has-jargon irods)
+           (jargon/get-paths irods uuids))) uuids])))
```

Comment:
I'd expect these to do something with the `cache?` variable. In particular, as currently written, this doesn't really use the functionality of `cached-or-get` much at all. If there's a cached jargon value but the icat is enabled, it will re-fetch the paths from the icat, since there's no guard on the first handler function only making real (non-cache) calls when `cache?` is true.

I'm not completely sure the best way to handle this case with these multi-fetch functions, though. I suppose ideally we'd allow each handler to return partial results, but maybe the effect would be mostly identical if it returns a cached value if and only if it can completely resolve it.

Reply:
I struggled with this quite a bit when I first started working on this, and I admittedly punted at the time. There's a bit of a mismatch between the multi-fetch functions and the single fetch functions that I wasn't quite sure how to resolve at first. Currently, the check for cached items happens in `cached-or-retrieved-values`, which is probably where most of the mismatch is coming from. Now that I've had some more time to wrap my head around this, I think we could probably resolve it by writing a multi-fetch version of `cached-or-get` that tries function calls in the same order on a potentially progressively smaller set of unresolved IDs. Then that function can return either when the set of unresolved IDs is empty or all of the lookup methods have been attempted.

That's a big enough change that I think I'd like to dedicate a separate pull request to it. That will give also give me some time to work on other tasks for a while without feeling like this one is looming. I'll add a comment and create a Jira issue for this.

Thanks for mentioning this. I meant to come back to it and forgot.

Reply:
I added the comment and created the Jira issue. Hopefully I can resume work on this in a couple of weeks.